### PR TITLE
Act on the readable subset of applications when appropriate

### PR DESCRIPTION
**Pull request by mpolden**
@jonmv

### CODE REVIEW

**Comment by mpolden** [comment]
Please confirm if we have the same views on what's appropriate. 🙂

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/ControllerTest.java [hunk 1]
```diff
@@ -939,4 +941,25 @@ public void testChangeEndpointCluster() {
...
+        db.curator().set(Path.fromString("/controller/v1/applications/t2:a2"), new byte[]{(byte) 0xDE, (byte) 0xAD});
+        assertEquals(1, tester.applications().readable().size());
+
+        try {
+            tester.applications().asList();
+            fail("Expected exception");
+        } catch (Exception ignored) {
+        }
+    }
+
```
Comment (jonmv) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/ControllerTest.java]: Perhaps add a test with a deployment scenario, where one app is unreadable?

**Review by jonmv — APPROVED** [review]
Looks good. A bit scary, so I'd like some more testing.

**Comment by mpolden** [comment]
PTAL at two last commits.

**Review by jonmv — APPROVED** [review]
Great!

**Pull request merged**